dask_glm/normal.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import absolute_import, division, print_function
+
+from dask_glm.utils import dot
+
+
+try:
+    from numba import jit
+except ImportError:
+    def jit(*args, **kwargs):
+        def _(func):
+            return func
+        return _
+
+
+def loglike(Xbeta, y):
+    return ((y - Xbeta)**2).sum()
+
+
+def pointwise_loss(beta, X, y):
+    beta, y = beta.ravel(), y.ravel()
+    Xbeta = X.dot(beta)
+    return loglike(Xbeta, y)
+
+
+def pointwise_gradient(beta, X, y):
+    beta, y = beta.ravel(), y.ravel()
+    Xbeta = X.dot(beta)
+    return gradient(Xbeta, X, y)
+
+
+def gradient(Xbeta, X, y):
+    return 2 *  dot(X.T, Xbeta) - 2 * dot(X.T, y)
+
+
+def hessian(Xbeta, X):
+    return dot(X.T, X)
```

Comment:
@mcg1969 Backtracking on my "all static methods" comment, I agree that preserving state actually makes sense here; for example, with a Gaussian / normal family, the Hessian only should be computed *once* because it is independent of `beta`.  Similarly, `p = sigmoid(Xbeta)` is re-used across gradient / Hessian calculations in the logistic model.

@mrocklin Once a log-likelihood is chosen, the gradient / hessian are determined for you -- which solver you use is *fairly* independent of family, as long as the model has enough derivatives for the solver to function properly (example: adding in `l1` regularization restricts you to `proximal_grad` and `admm` for the moment).